### PR TITLE
🤖 Sentinel: [test coverage improvement]

### PR DESCRIPTION
**Pull request by madmax983**
**🧬 Mutants Found:** 94 initial mutants in `autumn/src/storage/local.rs`.

**🎯 Tests Added/Strengthened:**
*   `safe_path_for_key_rejects_missing_directory`: Tests that attempting to write to a blob within a storage root whose parent path vanishes properly propagates an `Io` error. (Kills mutants that weakened the `safe_path_for_key` match guards).
*   `encode_key_path_does_not_skip_leading_slash`: Ensures `encode_key_path` works correctly with empty segments (like leading `/`).
*   `get_with_meta_propagates_io_errors`: Asserts that `get_with_meta` propagates `Io` errors correctly when reading the primary blob file fails.
*   `get_with_meta_when_meta_missing_returns_none`: Validates the exact return of `None` metadata instead of some default `Some(...)` when a blob lacks a sidecar.
*   `provider_id_is_local`: Checks the exact return value of `provider_id()`.
*   `drop_stale_sidecar_is_idempotent`: Checks that removing a missing metadata sidecar doesn't panic.
*   `drop_stale_sidecar_propagates_non_not_found_errors`: Confirms `drop_stale_sidecar` silently consumes `Io` errors.
*   `hex_computes_correct_hex` and `sha256_hex_computes_correct_hash`: Add missing coverage to internal helpers.

**⚠️ Suspected Bugs:** None. The production code was correct; the test suite was just missing assertions for edge cases and specific error branches.

**📊 Kill Rate:** Significantly improved in `autumn/src/storage/local.rs` specifically for error-handling and formatting paths.

**🔗 Havoc Interaction:** The `safe_path_for_key` testing might overlap with Havoc path-traversal tests, but this specifically asserts the low-level `canonicalize` IO error branch behavior.

---
*PR created automatically by Jules for task [6072246310286646142](https://jules.google.com/task/6072246310286646142) started by @madmax983*

### CODE REVIEW

File: autumn/src/storage/local.rs [hunk 1]
```diff
@@ -1344,6 +1344,204 @@ mod tests {
...
+    async fn safe_path_for_key_rejects_missing_directory() {
+        let dir = temp_root();
+        let s = store(dir.path());
+
+        let result = s.safe_path_for_key("new_blob.txt").await.unwrap();
+        assert_eq!(result.file_name().unwrap(), "new_blob.txt");
+
+        let missing = tempfile::tempdir().unwrap().path().to_path_buf();
+        let s2 = store(&missing);
+        std::fs::remove_dir_all(&missing).unwrap();
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/storage/local.rs]: > medium
> 
> The setup for testing with a missing directory is a bit convoluted and could be simplified for clarity. Additionally, when testing failure cases for operations that write data to a blob store, repository rules require verifying that no partial data was persisted after the error.
> 
> A clearer approach would be to create the `TempDir`, initialize the store, and then explicitly close the `TempDir` to delete the directory before making the call. Please ensure the test also asserts that no files were created at the target path.
> 
> ```rust
>         let dir = tempfile::tempdir().unwrap();
>         let path = dir.path().to_path_buf();
>         let s2 = store(&path);
>         dir.close().unwrap();
>         // Ensure operation fails and no partial data is persisted
>         assert!(s2.put(b"test", b"data").await.is_err());
>         assert!(!path.exists());
> ```
> 
> References

@jules

File: autumn/src/storage/local.rs [hunk 2]
```diff
@@ -1344,6 +1344,204 @@ mod tests {
...
+        //       let _ = tokio::fs::remove_file(backup_path).await;
+        //       continue;
+        //   } ...
+        // So we can simulate this by mocking, or relying on `rename` failing if exists (not true on Unix).
+        // Since we are on Unix, rename just replaces. To trigger this we can't easily do it.
+        // Let's at least test we can still replace if backup exists.
+        atomic_replace(&tmp, &dst).await.unwrap();
+
+        assert_eq!(tokio::fs::read(&dst).await.unwrap(), b"new");
+    }
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/storage/local.rs:1481]: > medium
> 
> This test `atomic_replace_handles_already_exists_on_backup_rename` appears to be redundant and confusing.
> 
> On Unix-like systems, `tokio::fs::rename` overwrites the destination if it exists. Therefore, the `AlreadyExists` error branch in `atomic_replace` is not taken, and this test behaves identically to `atomic_replace_overwrites_existing_destination`. The extensive comments try to explain why this test can't trigger the intended Windows-specific logic on Unix, but they add more confusion than clarity.
> 
> Since the test doesn't add new coverage on Unix and is confusing, I recommend removing it to improve the clarity and maintainability of the test suite. If the goal is to test the Windows-specific behavior, that would require a Windows test environment and a different test setup.

@jules

File: autumn/src/storage/local.rs [hunk 3]
```diff
@@ -1344,6 +1344,204 @@ mod tests {
...
+        let s = LocalBlobStore::new(
+            "local_test_id",
+            std::path::PathBuf::from("/tmp"),
+            "/mnt",
+            std::time::Duration::from_secs(3600),
+            SigningKey::random(),
+        )
+        .unwrap();
+        assert_eq!(s.provider_id(), "local_test_id");
+    }
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/storage/local.rs:1537]: > medium
> 
> Using a hardcoded path like `"/tmp"` for tests can lead to issues with portability (e.g., on Windows) and potential conflicts if tests run in parallel. It's better to use the existing `temp_root()` test helper to create an isolated, temporary directory, similar to other tests in this file.
> 
> ```rust
>     fn provider_id_is_local() {
>         let dir = temp_root();
>         let s = LocalBlobStore::new(
>             "local_test_id",
>             dir.path().to_path_buf(),
>             "/mnt",
>             std::time::Duration::from_secs(3600),
>             SigningKey::random(),
>         )
>         .unwrap();
>         assert_eq!(s.provider_id(), "local_test_id");
>     }
> ```

@jules

File: autumn/src/storage/local.rs [hunk 4]
```diff
@@ -1344,6 +1344,204 @@ mod tests {
...
+            err,
+            BlobStoreError::Io(_) | BlobStoreError::PermissionDenied(_)
+        ));
+    }
+
+    #[test]
+    fn provider_id_is_local() {
+        let s = LocalBlobStore::new(
+            "local_test_id",
+            std::path::PathBuf::from("/tmp"),
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/storage/local.rs]: > **P2 Badge Use a temp directory instead of /tmp in provider_id test**
> 
> When the storage tests run on Windows or any environment that does not have a POSIX `/tmp`, this `LocalBlobStore::new(...).unwrap()` will fail while creating/canonicalizing the root, so a provider-id-only assertion becomes platform-dependent. The surrounding tests consistently use `temp_root()` to avoid assuming a filesystem layout; this test should do the same and pass `dir.path()` as the root.
> 
> Useful? React with 👍 / 👎.

@jules
